docs/src/main/sphinx/develop/table-functions.rst

```diff
@@ -14,7 +14,8 @@ through implementing dedicated interfaces.
 Table function declaration
 --------------------------
 
-To declare a table function, you need to subclass ``ConnectorTableFunction``.
+To declare a table function, you need to implement ``ConnectorTableFunction``
+by subclassing ``AbstractConnectorTableFunction``.
```

Comment:
This should say that AbstractConnectorTableFunction is helpful.
We shouldn't _require_ extending from the class. The interface is the contract.

When thinking more about this, I am thinking what are these checks that `AbstractConnectorTableFunction` does. What would happen if a connector implemented the interface directly. Maybe the checks should be moved to `TableFunctionRegistry`?